src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -297,12 +306,69 @@ public InnReachResponseDTO recall(String trackingId, String centralCode, RecallD
     return success();
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var hold = transaction.getHold();
+    var loan = requestService.getLoan(hold.getFolioLoanId());
+    var existingDueDate = loan.getDueDate();
+    var requestedDueDate = new Date(borrowerRenew.getDueDateTime());
+
+    if (transaction.getHold().getFolioLoanId() != null) {
+      try {
+        var renewedLoan = renewLoan(transaction);
+        var calculatedDueDate = renewedLoan.getDueDate();
+        if (calculatedDueDate.after(requestedDueDate) || calculatedDueDate.equals(requestedDueDate)) {
+          transaction.setState(BORROWER_RENEW);
+        } else {
+          recallRequestToCentralSever(transaction, existingDueDate);
+        }
+      } catch (Exception e) {
+        if (existingDueDate.before(requestedDueDate)) {
+          recallRequestToCentralSever(transaction, existingDueDate);
+        } else {
+          throw new CirculationException("Failed to renew loan: " + e.getMessage(), e);
+        }
+      }
+    }
+
+    return success();
+  }
+
   private InnReachRecallUser getRecallUserForCentralServer(String centralCode) {
     return centralserverRepository.fetchOneByCentralCode(centralCode)
       .map(CentralServer::getInnReachRecallUser)
       .orElseThrow(() -> new EntityNotFoundException("Recall user is not set for central server with code = " + centralCode));
   }
 
+  private void recallRequestToCentralSever(InnReachTransaction transaction, Date existingDueDate) {
+    var trackingId = transaction.getTrackingId();
+    var centralCode = transaction.getCentralServerCode();
+
+    String uri = resolveD2irCircPath(D2IR_ITEM_RECALL_OPERATION, trackingId, centralCode);
+
+    var dueDateForRecallRequest = new HashMap<>();
+    dueDateForRecallRequest.put("dueDateTime", existingDueDate);
+    try {
+      innReachExternalService.postInnReachApi(centralCode, uri, dueDateForRecallRequest);
+      transaction.setState(RECALL);
+    } catch (Exception e) {
+      throw new CirculationException("Failed to recall request to central server: " + e.getMessage(), e);
+    }
+  }
+
+  private CheckOutResponseDTO renewLoan(InnReachTransaction transaction) {
+    var renewLoanRequestDTO = new RenewLoanRequestDTO();
+    renewLoanRequestDTO.setItemId(UUID.fromString(transaction.getHold().getItemId()));
```

Comment:
should probably be
transaction.getHold().getFolioItemId()

Reply:
yep, already fixed